data/json/npcs/missiondef.json

```diff
@@ -290,8 +290,23 @@
     "difficulty": 2,
     "value": 30000,
     "item": "safe_box",
-    "start": "place_deposit_box",
-    "end": "deposit_box",
+    "start": {
+      "update_mapgen": { "place_item": [ { "item": "safe_box", "x": [ 18, 21 ], "y": [ 4, 6 ], "amount": 1 } ] },
+      "effect": "follow_only",
+      "assign_mission_target": { "om_terrain": "bank", "om_terrain_match_type": "PREFIX", "search_range": 120, "random": true, "z": 0 }
+    },
+    "end": {
+      "effect": [
+        "stop_following",
+        {
+          "weighted_list_eocs": [
+            [ { "id": "get_katana" }, { "const": 2 } ],
+            [ { "id": "get_deagle_44" }, { "const": 3 } ],
+            [ { "id": "get_m4_carbine" }, { "const": 4 } ]
+          ]
```

Comment:
If I correctly understand how weighted list EOC works, I think this isn't quite a direct correlation to frequency in the code. Also, I'm bad at statistics, so I might be saying this wrong.
The code says, 33.3% of the time, end up with M4, 25% of the time, end up with katana, otherwise end up with desert eagle. Based on the NPC README, a higher value here corresponds to a higher likelihood. So I think desert eagle should be higher than M4? Ignoring the fact that the likelihood of katana and M4 are unrelated (there's a fancy stats word I'm forgetting), I think the desert eagle should be the result more frequently than the M4. If I were going to take a shot at roughly converting, I think I'd do - katana 5, M4 6, desert eagle 8

But anyone feel free to let me know if I'm totally misunderstanding how this attribute works :)

Reply:
I tried to copy the general order of reward chances from the C++ as best I could, in their highest order, with the M4 being first.

Reply:
std::string itemName = "deagle_44";
    if( one_in( 4 ) ) {
        itemName = "katana";
    } else if( one_in( 3 ) ) {
        itemName = "m4_carbine";
    }

I believe this translates to: start with item being desert eagle. roll a d4. if you get a 4, item changes to katana. roll a d3, if you get a 3, item changes to M4. give item to player.
So the likelihood of getting a desert eagle in the end is 3/4 x 2/3 = 6/12
so 3/12 chance of katana, 4/12 chance of M4, 6/12 chance of desert eagle
with some rounding... 
25/100, 33/100, 50/100
so I thiiink katana: 5, m4: 6, desert eagle: 10 (slight update after I actually attempted to math properly) would appropriately weight drop chances